### PR TITLE
Change dataCache variable structure.

### PR DESCRIPTION
**Pull request by 7kasper**

### CODE REVIEW

**Comment by 7kasper** [comment]
From #790

File: src/protocolsupport/protocol/utils/types/NetworkEntity.java [hunk 1]
```diff
@@ -56,12 +56,44 @@ public String toString() {
 	}
 
 	public static class DataCache {
-		public boolean firstMeta = true;
-		public byte baseMetaFlags;
+
+		private byte pcBaseFlags = 0;
```
Comment (Shevchik) [inline src/protocolsupport/protocol/utils/types/NetworkEntity.java]: pcBaseFlags -> baseFlags

File: src/protocolsupport/protocol/utils/types/NetworkEntity.java [hunk 2]
```diff
@@ -56,12 +56,44 @@ public String toString() {
...
+		public void setBaseFlag(int bitpos, boolean value) {
+			setBaseFlag(bitpos, value ? 1 : 0);
+		}
+
+		public void setBaseFlag(int bitpos, int value) {
+			baseFlags &= ~(1 << (bitpos - 1));
+			baseFlags |= (value << (bitpos - 1));
+		}
+
+		public void setPcBaseFlags(byte baseFlags) {
```
Comment (Shevchik) [inline src/protocolsupport/protocol/utils/types/NetworkEntity.java]: setBaseFlags

Comment (7kasper) [reply src/protocolsupport/protocol/utils/types/NetworkEntity.java]: derp

**Pull request merged**